runtime/pkg/rilltime/rilltime_test.go

```diff
@@ -0,0 +1,73 @@
+package rilltime
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Resolve(t *testing.T) {
+	now := parseTestTime(t, "2024-08-09T10:32:36Z")
+	maxTime := parseTestTime(t, "2024-08-06T06:32:36Z")
+	testCases := []struct {
+		timeRange string
+		start     string
+		end       string
+	}{
+		// Earliest = 2023-08-09T10:32:36Z, Latest = 2024-08-06T06:32:36Z, = Now = 2024-08-09T10:32:36Z
+		{`m : |s|`, "2024-08-09T10:32:00Z", "2024-08-09T10:32:36Z"},
```

Comment:
In the (resolved) comments on the doc, we discussed open intervals that end on a time boundary, but I don't think there's a test case for that here. Can you add it? I think an example could be:
```
		{`m : s`, "2024-08-09T10:32:00Z", "2024-08-09T10:32:37Z"},
```

Reply:
Wouldnt it be `2024-08-09T10:32:00Z` to `2024-08-09T10:32:36Z`?

Reply:
I believe it should be `...:37Z` – there was a long comment thread about it in the design doc, which seems to have been resolved without updating the main document. Here's one of the relevant comments (to find it, change the comment filter to include resolved comments in Notion):

<img width="328" alt="Screenshot 2025-01-10 at 11 16 44" src="https://github.com/user-attachments/assets/b3b396c2-76cc-4115-b172-afcea12f9468" />